blockchain/src/webhooks/index.ts

```diff
@@ -23,68 +26,93 @@ export function exit() {
-    if (blockHeight > currentBlock) {
-      if (blockHeight - minBlockConf < 1) {
-        log(`Current height is ${blockHeight}, waiting for ${env.MINIMUM_BLOCK_CONFIRMATIONS} blocks before processing...`);
-        return;
-      }
+  // Try again in 5 seconds if the next block isn't ready
+  if (height > highestBlock - MIN_BLOCK_CONF) {
+    blockScanTimeout = setTimeout(() => {
+      scanBlock(height);
+    }, 5000);
+    return;
+  }
 
-      const desiredBlock = currentBlock - minBlockConf;
-      try {
-        // Verbosity of 2 is full blocks
-        const block = await node.getblock(String(desiredBlock), 2);
-        log(`Processing block #${block.height}...`);
-        notifiers.forEach(n => n.onNewBlock && n.onNewBlock(block));
-        currentBlock++;
-        consecutiveBlockFailures = 0;
-      } catch(err) {
-        log(err.response ? err.response.data : err);
-        log(`Failed to fetch block ${desiredBlock}`);
-        consecutiveBlockFailures++;
-        if (consecutiveBlockFailures >= MAXIMUM_BLOCK_FAILURES) {
-          log('Maximum consecutive failures reached, exiting!');
-          process.exit(1);
-        }
-        else {
-          log('Attempting to fetch again shortly...');
-        }
-      }
+  // Process the block
+  try {
+    const block = await node.getblock(String(height), 2); // 2 == full blocks
+    log(`Processing block #${block.height}...`);
+    notifiers.forEach(n => n.onNewBlock && n.onNewBlock(block));
+    consecutiveBlockFailures = 0;
+  } catch(err) {
+    log(err.response ? err.response.data : err);
+    log(`Failed to fetch block ${height}, see above error`);
+    consecutiveBlockFailures++;
+    // If we fail a certain number of times, it's reasonable to
+    // assume that the blockchain is down, and we should just quit.
+    // TODO: Scream at sentry or something!
+    if (consecutiveBlockFailures >= MAXIMUM_BLOCK_FAILURES) {
+      log('Maximum consecutive failures reached, exiting!');
+      process.exit(1);
     }
-  }, 3000);
-}
+    else {
+      log('Attempting to fetch again shortly...');
+      await sleep(5000);
+    }
+  }
 
+  // Try next block
+  scanBlock(height + 1);
+}
 
 function initNotifiers() {
-  const send: Send = (route, method, payload) => {
-    console.log('About to send to', route);
-    axios.request({
-      method,
-      url: `${env.WEBHOOK_URL}${route}`,
-      data: payload,
-      headers: {
-        'Authorization': `Bearer ${env.API_SECRET_KEY}`,
-        'Content-Type': 'application/json',
-      },
-    })
-    .then((res) => {
-      if (res.status >= 400) {
-        console.error(`Webhook server responded to ${method} ${route} with status code ${res.status}`);
-        console.error(res.data);
-      }
-    })
-    .catch((err) => {
-      console.error(err);
-      console.error('Webhook server request failed! See above for details.');
-    });
-  };
-
   notifiers = initializeNotifiers();
   notifiers.forEach(n => n.registerSend(send));
```

Comment:
Now that `send` is a function that could stand alone, perhaps we should move it to its own file and just have each notifier import it?